layers/+lang/vala/packages.el

```diff
@@ -0,0 +1,47 @@
+;;; packages.el --- vala layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Steven Allen <steven@stebalien.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;;; Code:
+
+(defconst vala-packages
+  '(
+    vala-mode
+    yasnippet
+    (vala-snippets :toggle (configuration-layer/package-usedp 'yasnippet))
+    flycheck
+    (flycheck-vala :toggle (configuration-layer/package-usedp 'flycheck)
+                   :location (recipe
+                              :fetcher git
+                              :url "https://notabug.org/koz.ross/flycheck-vala.git")))
+  "The list of Lisp packages required by the vala layer.")
+
+(defun vala/post-init-flycheck ()
+  (spacemacs/add-flycheck-hook 'vala-mode))
+
+(defun vala/init-flycheck-vala ()
+  (use-package flycheck-vala
+    :defer t
+    :init (add-hook 'vala-mode-hook (lambda () (require 'flycheck-vala)))))
```

Comment:
I've changed this to use the `vala-mode-hook` instead of `flycheck-mode-hook` so that it gets loaded before file/directory local variables are read. Is this the correct way to do this?